tests/e2e.rs

```diff
@@ -200,3 +201,70 @@ async fn e2e_punish_publication_of_revoked_commit_transaction() {
         .await
         .unwrap();
 }
+
+#[tokio::test]
+async fn e2e_channel_collaborative_close() {
+    let tc_client = testcontainers::clients::Cli::default();
+    let bitcoind = Bitcoind::new(&tc_client, "0.19.1").unwrap();
+
+    let time_lock = 1;
+    let (initial_alice_balance, initial_bob_balance) = (Amount::ONE_BTC, Amount::ONE_BTC);
+
+    let create::Init {
+        alice,
+        alice_wallet,
+        bob,
+        bob_wallet,
+    } = create::Init::new(
+        &bitcoind,
+        initial_alice_balance,
+        initial_bob_balance,
+        time_lock,
+    )
+    .await;
+
+    let create::Final { alice, bob } = create::run(&alice_wallet, alice, &bob_wallet, bob).await;
+
+    alice_wallet
+        .0
+        .send_raw_transaction(alice.signed_TX_f.clone())
+        .await
+        .unwrap();
+
+    let close::Init { alice, bob } = close::Init::new(alice, bob);
+
+    let alice_final_address = alice_wallet.0.new_address().await.unwrap();
+    let bob_final_address = bob_wallet.0.new_address().await.unwrap();
+
+    let close::Final { alice, bob } =
+        close::run(alice, alice_final_address, bob, bob_final_address).unwrap();
+
+    let alice_closing_transaction = alice.into_transaction();
+    let bob_closing_transaction = bob.into_transaction();
+    assert_eq!(alice_closing_transaction, bob_closing_transaction);
+
+    // ugly wait for 1 block
+    tokio::time::delay_for(std::time::Duration::from_millis(2_000)).await;
```

Comment:
:question: What do we need to wait for?

Reply:
We need to wait the bitcoinharness to generate blocks. Otherwise the `TX_f` was not mined and the wallet still reflects the initial balance.

Reply:
Oh, true.